extensions/ql-vscode/src/query-history.ts

```diff
@@ -639,6 +640,15 @@ export class QueryHistoryManager extends DisposableObject {
     return this.treeDataProvider.getCurrent();
   }
 
+  async removeDeletedQueries() {
+    await Promise.all(this.treeDataProvider.allHistory.map(async (item) => {
+      if (item.t == 'local' && item.completedQuery && !(await fs.pathExists(item.completedQuery?.query.querySaveDir))) {
```

Comment:
Minor: No need to check if `item.completedQuery` is truthy if your are using `.?` later on.

```suggestion
      if (item.t == 'local' && !(await fs.pathExists(item.completedQuery?.query.querySaveDir))) {
```